File: BC-W4/Army/Spells/AquaSplash/AquaSplash.cpp
#include "AquaSplash.h"

AquaSplash::AquaSplash(int actionPoints, int manaCost, const std::string& spellName) : Spell(actionPoints, manaCost, spellName) {};
AquaSplash::~AquaSplash() {};

void AquaSplash::action(Unit* enemy) {
    enemy->takeMagicDamage(this->getActionPoints());
};

void AquaSplash::heal(Unit* unit) {
    throw InvalidHealException();
};

File: BC-W4/ArmyJava/src/spells/AquaSplash.java
package spells;

import exceptions.InvalidCastException;
import units.Unit;

public class AquaSplash extends Spell {
    public AquaSplash() {
        super("AquaSplash", 45, 65, true);
    }

    public AquaSplash(int actionPoints) {
        super("AquaSplash", 45, actionPoints, true);
    }

    public void action(Unit enemy) throws InvalidCastException {
        enemy.takeMagicDamage(this.actionPoints);
    }

    public void heal(Unit unit) throws InvalidCastException {
        throw new InvalidCastException();
    }

}
